internal/pkg/cli/pipeline_init.go

```diff
@@ -326,38 +392,62 @@ func (o *initPipelineOpts) parseBitbucketRepoDetails() error {
 	o.repoName = repoDetails.name
 	o.repoOwner = repoDetails.owner
 
-	if o.repoBranch == "" {
-		o.repoBranch = defaultBBBranch
-	}
 	return nil
 }
 
 func (o *initPipelineOpts) selectURL() error {
-	// Fetches and parses all remote repositories.
-	err := o.runner.Run("git", []string{"remote", "-v"}, exec.Stdout(&o.buffer))
+	repos, err := o.fetchAndParseURLs()
 	if err != nil {
-		return fmt.Errorf("get remote repository info: %w; make sure you have installed Git and are in a Git repository", err)
+		return fmt.Errorf("fetch and parse URLs: %w", err)
 	}
-	urls, err := o.parseGitRemoteResult(strings.TrimSpace(o.buffer.String()))
-	if err != nil {
-		return err
+
+	var formattedRepos []string
+	for url, name := range repos {
+		formattedRepos = append(formattedRepos, fmt.Sprintf("%s: %s", name, url))
```

Comment:
How about we make `%s: %s` to be a string type called maybe `pipelineRepo` with two methods `url()` and `shortName()` and a constructor method `newPipelineRepo()`?

Reply:
Instead of formatting strings for `SelectOne`, we could take advantage of this [`SelectOption`](https://github.com/aws/copilot-cli/blob/1ba3833f7cda22ac876902148db02881dd1a5a23/internal/pkg/term/prompt/select.go#L55):

```
var options []Option
for url, name := range repos {
   options = append(options, Option{Value: url, Hint: name})
}

url, err := o.prompt.SelectOption(
pipelineSelectURLPrompt, 
pipelineSelectURLHelpPrompt, 
options,
prompt.WithFinalMessage("Repository:"),
```

and the selected `url` would be just the url, without the short name.

Reply:
Ah, `SelectOne` is similar to what I was looking for, but the problem is that we need both the url (for parsing info out of) and the remoteName/shortname (to pass to `git branch`). That's why I added the remoteName in the first place-- we did not used to have that at all. What I need is a parseValueAndHintFromOptionFmt, but that would just do what I'm doing here. 🤔